Mathlib/LinearAlgebra/Matrix/Swap.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2024 Judith Ludwig, Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Christian Merten
+-/
+import Mathlib.LinearAlgebra.Matrix.GeneralLinearGroup.Defs
+import Mathlib.Data.Matrix.PEquiv
+
+/-!
+# Swap matrices
+
+A swap matrix indexed by `i` and `j` is the matrix that, when multiplying another matrix
+on the left (resp. on the right), swaps the `i`-th row with the `j`-th row
+(resp. the `i`-th column with the `j`-th column).
+
+Swap matrices are a special case of *elementary matrices*. For transvections see
+`Mathlib.LinearAlgebra.Matrix.Transvection`.
+
+## Implementation detail
+
+This is a thin wrapper around `(Equiv.swap i j).toPEquiv.toMatrix`.
+-/
+
+namespace Equiv
+
+open Matrix
+
+variable {α n m : Type*}
+
+lemma toPEquiv_toMatrix_mulVec_apply [Fintype n] [DecidableEq n]
+    [NonAssocSemiring α] (σ : m ≃ n) (i : m) (a : n → α) :
+    (σ.toPEquiv.toMatrix *ᵥ a) i = a (σ i) := by
+  induction' a using Pi.induction_add with f g hf hg x y
+  · simp
+  · simp [hf, hg, Matrix.mulVec_add]
+  · simp [Pi.single_apply]
+
+lemma toPEquiv_toMatrix_vecMul_apply [Fintype m] [DecidableEq n]
+    [NonAssocSemiring α] (σ : m ≃ n) (i : n) (a : m → α) :
+    (a ᵥ* σ.toPEquiv.toMatrix) i = a (σ.symm i) := by
+  classical
+  induction' a using Pi.induction_add with f g hf hg x y
+  · simp
+  · simp [hf, hg, Matrix.add_vecMul]
+  · simp only [single_vecMul, PEquiv.toMatrix_apply, Option.mem_def, toPEquiv_eq_some_iff, mul_ite,
+      mul_one, mul_zero, Pi.single_apply, symm_apply_eq]
+    simp_rw [eq_comm]
```

Comment:
Do you have an idea where these should go? Note that `Pi.induction_add` is not available in `Data.Matrix.PEquiv` because `Algebra.BigOperators.Pi` is not available.